### PR TITLE
Lint gondar with cpplint

### PR DESCRIPTION
**Pull request by kendase3**
Partially addresses OVER-8773

Notably, ignore src/gondar.cc for now.

If you merge kendall-lint-02(this PR: the lint fixes themselves) and kendall-cpplint-04(the ability to run make lint') into a local master branch and run 'make lint', you should get zero errors.

### CODE REVIEW

**Comment by kendase3** [comment]
the more controversial changes in here (there are quite a few boring changes) are the cplusplusification of casts in some still very C-like contexts, such as macros.  Personally I don't feel this helps with clarity but I'd rather minimize the linter config.

**Comment by kendase3** [comment]
the PR builder failed: it doesn't like some of my reinterpret_cast changes.  i suppose i should just ignore the C-style casts for now; i'll make that change on monday.

**Comment by kendase3** [comment]
@nicholasbishop also take a gander at this plx :basketball_man:

File: src/downloader.cc [hunk 1]
```diff
@@ -31,7 +31,7 @@ DownloadManager::DownloadManager(QObject* parent)
     : QObject(parent), error(false), downloadedCount(0), totalCount(0) {}
 
 void DownloadManager::append(const QStringList& urlList) {
-  foreach (QString url, urlList)
+  foreach(QString url, urlList)
```
Comment (nicholasbishop) [inline src/downloader.cc]: Might as well switch this to a C++11-style for loop

File: src/downloader.cc [hunk 2]
```diff
@@ -31,7 +31,7 @@ DownloadManager::DownloadManager(QObject* parent)
     : QObject(parent), error(false), downloadedCount(0), totalCount(0) {}
 
 void DownloadManager::append(const QStringList& urlList) {
-  foreach(QString url, urlList)
+  for (QString url : urlList)
```
Comment (nicholasbishop) [inline src/downloader.cc]: `const auto url&`

**Pull request merged**